### PR TITLE
XPath 3.1 support...

### PR DESCRIPTION
**Pull request by RackerWilliams**
...it's time.

### CODE REVIEW

File: core/src/main/scala/com/rackspace/com/papi/components/checker/step/StepHandler.scala [hunk 1]
```diff
@@ -755,7 +755,7 @@ class StepHandler(var contentHandler : ContentHandler, val config : Config) exte
       expression = XPathExpressionPool.borrowExpression(_match, context, version)
     } catch {
       case spe : SAXParseException => throw spe
-      case e : Exception => throw new SAXParseException ("Error while compiling XPath expression", locator, e)
+      case e : Exception => throw new SAXParseException ("Error while compiling XPath expression: "+e.getMessage(), locator, e)
```
Comment (RackerWilliams) [inline core/src/main/scala/com/rackspace/com/papi/components/checker/step/StepHandler.scala]: I noticed the exception was listing useful human readable information as to why things failed, when handling JSON XPath and display the message there. So extend to support regular XPath calls as well.

File: core/src/main/scala/com/rackspace/com/papi/components/checker/util/XPathExpressionPool.scala [hunk 2]
```diff
@@ -104,9 +104,10 @@ private class XPathExpressionFactory(private val expression : String, private va
 }
 
 
-private class XPath2ExpressionFactory(private val expression : String, private val nc : NamespaceContext) extends XPathExpressionFactory(expression, nc) {
+private class XPathNExpressionFactory(private val expression : String, private val nc : NamespaceContext, private val version : Int) extends XPathExpressionFactory(expression, nc) {
   override def makeObject = {
     val xpath = (new net.sf.saxon.xpath.XPathFactoryImpl()).newXPath()
```
Comment (RackerWilliams) [inline core/src/main/scala/com/rackspace/com/papi/components/checker/util/XPathExpressionPool.scala]: The complexity is that XPathNExpressionFactory supports versions 2.0 to 3.1 is backed by Saxon.  XPath 1.0 is backed by the JRE implementation.  Interestingly Saxon doesn't support XPath 1.0, which is why we have two factories.  2.0 is not enteirly backwards compatible with 1.0 in some extream edge cases, so I'm just being safe there.

File: core/src/test/scala/com/rackspace/com/papi/components/checker/ValidatorWADLPlainParamSuite.scala [hunk 3]
```diff
@@ -931,6 +1240,14 @@ class ValidatorWADLPlainParamSuite extends BaseValidatorSuite {
     validator_XSDElementContentPlainOpt2.validate(request("GET","/c"),response,chain)
   }
 
+  test ("GOT on /c should succeed on validator_XSDElementContentPlainOpt30") {
```
Comment (RackerWilliams) [inline core/src/test/scala/com/rackspace/com/papi/components/checker/ValidatorWADLPlainParamSuite.scala]: Oh man, I'll do a PR to fix those 😳

File: core/src/test/scala/com/rackspace/com/papi/components/checker/wadl/WADLCheckerOptSpec.scala [hunk 4]
```diff
@@ -209,10 +209,172 @@ class WADLCheckerOptSpec extends BaseCheckerSpec {
...
       assert (checker, "count(/chk:checker/chk:step[@type='XPATH' and @match='/atom:entry/@only_usage']) = 1")
-      assert (checker, "count(/chk:checker/chk:step[@type='XSL']) = 2")
+      assert (checker, "count(/chk:checker/chk:step[@type='XSL' and @version='1']) = 2")
       assert (checker, "count(/chk:checker/chk:step[@type='XSL']//xsl:when[@test='/atom:entry']) = 2")
     }
 
+
+    scenario ("The sharedXPathWADL is processed checking wellformness, XSD, elemnts, and plain parameters, with remove dups and joinpath optimizations (XPath 2.0)") {
+      Given("the sharedXPathWADL")
+      When("The WADL is transalted")
```
Comment (RackerWilliams) [inline core/src/test/scala/com/rackspace/com/papi/components/checker/wadl/WADLCheckerOptSpec.scala]: Ugh, I'll get those to.

File: core/src/main/scala/com/rackspace/com/papi/components/checker/util/XPathExpressionPool.scala (hunk 2 above)
Comment (wdschei) [inline core/src/main/scala/com/rackspace/com/papi/components/checker/util/XPathExpressionPool.scala]: I get that, I was just saying we could consolidate all the `version match` blocks into the `private` factory implementation itself so that it was streamlined everywhere else.

Comment (wdschei) [inline core/src/main/scala/com/rackspace/com/papi/components/checker/util/XPathExpressionPool.scala]: No biggie really. It is more of a potato / potato thing than an actual functional thing.
